src/test/java/com/autosale/shop/service/UserServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package com.autosale.shop.service;
+
+
+import com.autosale.shop.model.User;
+import com.autosale.shop.model.UserRole;
+import com.autosale.shop.repository.UserRepository;
+import com.autosale.shop.repository.impl.UserRepositoryImpl;
+import com.autosale.shop.service.impl.UserServiceImpl;
+import net.bytebuddy.utility.RandomString;
+import org.jooq.exception.DataAccessException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class UserServiceTest {
+    private final UserRepository repository = Mockito.mock(UserRepositoryImpl.class);
+    private final UserService userService = new UserServiceImpl(repository);
+
+    List<User> users = new ArrayList<>();
+
+    @BeforeEach
+    void setUp() {
+        Random rand = new Random();
+        for (int i = 0; i < 3; i++) {
+            users.add(new User(rand.nextInt(100), RandomString.make(50), RandomString.make(25), rand.nextBoolean() ? UserRole.USER : UserRole.ADMIN));
```

Comment:
usually, we use UUID.randomUUID().toString() for generation of random string

Reply:
even if it doesn't fit the parameter max size?

Reply:
Do you have max size validation in User model?

Reply:
I got your point, we don't need to suit objects to the database model without internal validation